modular_citadel/code/modules/arousal/organs/genitals.dm

```diff
@@ -474,10 +474,28 @@
 			colourcode = S.color_src
 
 			if(G.slot == "belly") // GS13
-				genital_overlay.icon = 'hyperstation/icons/obj/genitals/belly.dmi'
-				genital_overlay.icon_state = "belly_[size]"
-				genital_overlay.layer = -UNDER_BACK_LAYER
-				colourcode = "belly_color"
+				// Change belly sprite and size based on current fullness
+				switch(H.fullness)
+					if(0 to FULLNESS_LEVEL_BLOATED)
+						genital_overlay.icon = 'hyperstation/icons/obj/genitals/belly.dmi'
+						genital_overlay.icon_state = "belly_[size]"
+						genital_overlay.layer = -UNDER_BACK_LAYER
+						colourcode = "belly_color"
```

Comment:
```suggestion
						genital_overlay.icon_state = "belly_[size]"
```

I think that having the icon, layer, and colorcode be set every-time might be redundant, but I am not entirely sure.

Reply:
Indeed, it needed cleaning up.
I extracted the default vars to the top, and the switch only changes the applicable vars